libs/math/include/math/matrix_operations.hpp

```diff
@@ -705,12 +706,7 @@ void ReduceMax(ArrayType const &obj1, SizeType axis, ArrayType &ret)
   using DataType = typename ArrayType::Type;
   ret.Fill(std::numeric_limits<DataType>::min());
 
-  Reduce(axis,
-         [](const DataType &x, DataType &y) {
-           if (x > y)
-             y = x;
-         },
-         obj1, ret);
+  Reduce(axis, [](const DataType &x, DataType &y) { y = std::max(x, y); }, obj1, ret);
```

Comment:
presumably this will not work for fixed point types? or does std::max just call the operator> if it doesn't know what to do?
please make sure this is covered with a test.